### PR TITLE
SAMBRO: admin subscriptions

### PR DESCRIPTION
**Pull request by biplovbhandari**
@flavour Please merge

### CODE REVIEW

**Comment by biplovbhandari** [comment]
@flavour Please review/merge

**Comment by biplovbhandari** [comment]
@flavour Please merge

File: modules/templates/SAMBRO/config.py [hunk 1]
```diff
@@ -271,6 +271,42 @@ def custom_prep(r):
...
+        # Custom postp
+        standard_postp = s3.postp
+        def custom_postp(r, output):
+            # Call standard postp
+            if callable(standard_postp):
+                output = standard_postp(r, output)
+
+            if r.interactive and isinstance(output, dict):
+                # Modify Update Button
+                s3_action_buttons = S3CRUD.action_buttons
```
Comment (flavour) [inline modules/templates/SAMBRO/config.py]: Why the intermediate variable?

File: modules/templates/SAMBRO/controllers.py [hunk 2]
```diff
@@ -339,24 +381,69 @@ def _manage_subscriptions(self, resources, filters):
...
 
-        # Filters
-        filter_form = S3FilterForm(filters, clear=False)
-        fieldset = FIELDSET(filter_form.fields(None,
-                                               subscription["get_vars"]),
-                            _id="subscription-filter-form")
-        form.append(fieldset)
+        row = db(gtable.role == "Alert Editor").select(gtable.id,
+                                                       limitby=(0, 1)).first()
+        ae_id = row.id
```
Comment (flavour) [inline modules/templates/SAMBRO/controllers.py]: There should be some kind opf documentation...even if just a comment to explain that we're using the Alert Editor role as the owned_by_group to distinguish Admin-controlled subscriptions

File: modules/templates/SAMBRO/controllers.py [hunk 3]
```diff
@@ -515,28 +605,288 @@ def _manage_subscriptions(self, resources, filters):
...
+            bulk_user_pe_id = auth.s3_bulk_user_pe_id
+            pr_group_table = s3db.pr_group
+            define_resource = s3db.resource
+            base_query = (stable.deleted != True)
+            if user_ids is not None:
+                # Admin Subscriptions
+                if group_ids is not None and \
+                   len(group_ids) > 0 and \
+                   len(user_ids) > 0:
+                    # Both selection
```
Comment (flavour) [inline modules/templates/SAMBRO/controllers.py]: I don't understand this comment?

Comment (flavour) [inline modules/templates/SAMBRO/controllers.py]: Ah: "Both Users & Groups selected"

File: modules/templates/SAMBRO/controllers.py [hunk 4]
```diff
@@ -515,28 +605,288 @@ def _manage_subscriptions(self, resources, filters):
...
+                                        id=subscription_id).delete()
+                        define_resource("pr_filter",
+                                        id=row.filter_id).delete()
+                redirect(URL(c="pr", f="subscription"))
             else:
-                self._remove_sync(properties)
-                db(stable.pe_id == current.auth.user.pe_id).update(comments=None)
+                # Normal user
+                success_subscription = self._update_subscription(subscription)
+                current_pe_id = auth.user.pe_id
```
Comment (flavour) [inline modules/templates/SAMBRO/controllers.py]: I don't see the value of this intermediate variable.
auth.user, however would be useful to bring into scope as used twice.

File: modules/templates/SAMBRO/menus.py [hunk 5]
```diff
@@ -75,8 +75,13 @@ def menu_modules(cls):
                                   ])
                 elif auth.s3_has_role("ALERT_EDITOR") or \
                      auth.s3_has_role("ALERT_APPROVER"):
-                    menus_.append(homepage("cap"),
-                                  )
+                    menus_.extend([homepage("cap"),
+                                  homepage("pr", name=current.T("Manage"),
+                                           link=False)(
+                        MM("Manage Recipients", c="pr", f="subscription"),
+                        MM("Manage Group for recipients", c="pr", f="group"),
```
Comment (flavour) [inline modules/templates/SAMBRO/menus.py]: Groups plural?

Comment (flavour) [inline modules/templates/SAMBRO/menus.py]: How about "Manage Recipient Groups"

File: modules/templates/SAMBRO/menus.py [hunk 6]
```diff
@@ -103,11 +108,20 @@ def menu_auth(cls, **attr):
         else:
             # Logged-in
             user_id = auth.s3_logged_in_person()
+            if auth.s3_has_role("ALERT_EDITOR") or \
+               auth.s3_has_role("ALERT_APPROVER"):
```
Comment (flavour) [inline modules/templates/SAMBRO/menus.py]: could bring s3_has_role into scope before the if

File: modules/templates/SAMBRO/menus.py [hunk 7]
```diff
@@ -103,11 +108,20 @@ def menu_auth(cls, **attr):
...
+                                           f="index",
+                                           m="subscriptions",
+                                           )
+                
             menu_auth = MM(auth.user.email, link=False, right=True)(
                            MM("Edit Profile", c="pr", f="person", args=[user_id]),
                            MM("Details", c="default", f="user", m="profile"),
-                           MM("Notification Settings", c="default", f="index",
-                              m="subscriptions"),
+                           notification_settings,
```
Comment (flavour) [inline modules/templates/SAMBRO/menus.py]: I still don't like this being the location for these options.
This is a Personal menu...not a menu for Admin tasks.
I suggest moving these to either the CAP module or Admin.
I think the CAP module menu makes most sense to me...it already has many Admin functions in

**Comment by flavour** [comment]
Looking a lot better now...some minor tweaking to do & then we can merge & try it out :)

**Comment by biplovbhandari** [comment]
@flavour Done...

**Pull request closed without merging**